nexus_motion_planner/test/test_motion_plan_cache.py

```diff
@@ -0,0 +1,398 @@
+            'Fetched plan has correct planning time',
+            'Fetched plan has correct fraction',
+        ]),
+        ('test_cartesian_plans.put_first.fetch_higher_fraction', [
+            'Fetch all returns empty',
+            'Fetch best plan is nullptr',
+        ]),
+        ('test_cartesian_plans.put_first.fetch_lower_fraction', [
+            'Fetch all returns one',
+            'Fetch best plan is not nullptr',
+            'Fetched plan on both fetches match',
+            'Fetched plan matches original',
+            'Fetched plan has correct execution time',
+            'Fetched plan has correct planning time',
+            'Fetched plan has correct fraction',
+        ]),
+
+        ('test_cartesian_plans.put_worse', [
+            'Put worse plan, no delete_worse_plans, not ok',
+            'One plan in cache',
+        ]),
+
+        ('test_cartesian_plans.put_better', [
+            'Put better plan, no delete_worse_plans, ok',
+            'Two plans in cache',
+        ]),
+        ('test_cartesian_plans.put_better.fetch_sorted', [
+            'Fetch all returns two',
+            'Fetch best plan is not nullptr',
+            'Fetched plan on both fetches match',
+            'Fetched plan matches original',
+            'Fetched plan has correct execution time',
+            'Fetched plan has correct planning time',
+            'Fetched plan has correct fraction',
+            'Fetched plans are sorted correctly',
+        ]),
+        ('test_cartesian_plans.put_better_delete_worse_plans', [
+            'Put better plan, delete_worse_plans, ok',
+            'One plan in cache',
+        ]),
+        ('test_cartesian_plans.put_better_delete_worse_plans.fetch', [
+            'Fetch all returns one',
+            'Fetch best plan is not nullptr',
+            'Fetched plan on both fetches match',
+            'Fetched plan matches original',
+            'Fetched plan has correct execution time',
+            'Fetched plan has correct planning time',
+            'Fetched plan has correct fraction',
+        ]),
+
+        ('test_cartesian_plans.put_different_req', [
+            'Put different plan req, delete_worse_plans, ok',
+            'Two plans in cache',
+        ]),
+        ('test_cartesian_plans.put_different_req.fetch', [
+            'Fetch all returns one',
+            'Fetch best plan is not nullptr',
+            'Fetched plan on both fetches match',
+            'Fetched plan matches original',
+            'Fetched plan has correct execution time',
+            'Fetched plan has correct planning time',
+            'Fetched plan has correct fraction',
+        ]),
+    ]
+
+    for prefix, labels in test_cases:
+        for label in labels:
+            process_tools.assert_output_sync(
+                launch_context,
+                motion_cache_test_runner_node,
+                validate_stream(f"[PASS] {prefix}: {label}"),
+                timeout=10
+            )
```

Comment:
Unless I'm missing something, I don't think we need to scan the output, the main test exits with non-zero if anything fails, all we need is to catch that and report the result accordingly.

imo this is also not good design, it is repeating the test case and labels, making modification unnecessarily errior-prone and tedious. It may also break if the uut code prints something to stdout.